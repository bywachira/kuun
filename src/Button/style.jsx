import styled from 'styled-components'

export const Button = styled.button.attrs((props) => ({
  onClick: props.onClick
}))`
  font-size: 17px;
  appearance: none;
  white-space: nowrap;
  word-break: keep-all;
  cursor: pointer;
  line-height: 1;
  position: relative;
  text-align: center;
  box-shadow: rgba(0, 0, 0, 0.02) 0px 1px 2px;
  opacity: 1;
  font-weight: 500;
  display: inline-block;
  color: ${(props) =>
    props.type === 'primary' || props.type === 'danger'
      ? 'rgb(255, 255, 255)'
      : 'rgb(68, 71, 74)'};
  padding: ${(props) =>
    props.size === 'large'
      ? '16px 64px'
      : props.size === 'small'
      ? '4px 16px'
      : props.size === 'medium'
      ? '8px 32px'
      : '2px 8px'};
  border-radius: ${(props) => (props.shape === 'pill' ? '32px' : '4px')};
  transition: all 0.2s ease-out 0s;
  text-decoration: none;
  background: ${(props) =>
    props.type === 'primary'
      ? 'linear-gradient(to top, #21bf73, #28df99)'
      : props.type === 'danger'
      ? 'linear-gradient(to top, #fd5e53, #fa697c)'
      : 'linear-gradient(to top, rgb(250, 251, 252), rgb(255, 255, 255))'};
  border-width: 1px;
  border-style: solid;
  border-color: ${(props) =>
    props.type === 'primary'
      ? '#21bf73'
      : props.type === 'danger'
      ? '#fa697c'
      : 'rgb(233, 236, 240)'};
  border-image: initial;

  &:hover {
    opacity: 0.9;
  }
`
