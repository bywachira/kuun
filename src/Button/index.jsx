import React from 'react'
import * as Styles from './style'

function Button({ onClick, type, shape, size, children }) {
  return (
    <Styles.Button onClick={onClick} type={type} shape={shape} size={size}>
      {children}
    </Styles.Button>
  )
}

export default Button
