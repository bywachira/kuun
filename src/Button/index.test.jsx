import React from 'react'
import enzyme, { mount } from 'enzyme'
import toJSON from 'enzyme-to-json'
import adapter from 'enzyme-adapter-react-16'
import Button from '.'

enzyme.configure({ adapter: new adapter() })

describe('#BUTTON', () => {
  it('should render properly', () => {
    const component = mount(
      <Button
        onClick={() => {}}
        type='default'
        shape='pill'
        size='large'
        children='children'
      />
    )

    expect(toJSON(component)).toMatchSnapshot()
  })
})
