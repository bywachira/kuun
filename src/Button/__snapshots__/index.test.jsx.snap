// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#BUTTON should render properly 1`] = `
<Button
  onClick={[Function]}
  shape="pill"
  size="large"
  type="default"
>
  <styled.button
    onClick={[Function]}
    shape="pill"
    size="large"
    type="default"
  >
    <button
      className="sc-bdnylx ectldl"
      onClick={[Function]}
      shape="pill"
      size="large"
      type="default"
    >
      children
    </button>
  </styled.button>
</Button>
`;
